libs/common/Networking/NetworkHandler.cs

```diff
@@ -271,11 +271,11 @@ async Task AuthenticateAsClientAsync(SslClientAuthenticationOptions sslClientOpt
         /// On network receive
         /// </summary>
         /// <param name="bytesTransferred">Number of bytes transferred</param>
-        public unsafe void OnNetworkReceive(int bytesTransferred)
+        public async Task OnNetworkReceiveAsync(int bytesTransferred)
```

Comment:
Using `Task` here will cause a class instance allocation per call, which is very expensive. It seems even the non-TLS paths will cause this. We should use the `ValueTask` struct here.

Reply:
Have changed to use ValueTask